src/Components/Aspire.Pomelo.EntityFrameworkCore.MySql/AspireEFMySqlExtensions.cs

```diff
@@ -147,18 +96,143 @@ void ConfigureDbContext(IServiceProvider serviceProvider, DbContextOptionsBuilde
+            }
+        }
+    }
+
+    /// <summary>
+    /// Configures retries, health check, logging and telemetry for the <see cref="DbContext" />.
+    /// </summary>
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="DbContext"/> is not registered in DI.</exception>
+    public static void EnrichMySqlDbContext<[DynamicallyAccessedMembers(RequiredByEF)] TContext>(
+            this IHostApplicationBuilder builder,
+            Action<PomeloEntityFrameworkCoreMySqlSettings>? configureSettings = null) where TContext : DbContext
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = GetDbContextSettings<TContext>(builder);
+
+        configureSettings?.Invoke(settings);
+
+        ConfigureRetry();
+
+        ConfigureInstrumentation<TContext>(builder, settings);
+
+        void ConfigureRetry()
+        {
+            if (!settings.Retry)
+            {
+                return;
             }
+
+            // Resolving DbContext<TContextService> will resolve DbContextOptions<TContextImplementation>.
+            // We need to replace the DbContextOptions service descriptor to inject more logic. This won't be necessary once
+            // Aspire targets .NET 9 as EF will respect the calls to services.ConfigureDbContext<TContext>(). c.f. https://github.com/dotnet/efcore/pull/32518
+
+            var oldDbContextOptionsDescriptor = builder.Services.FirstOrDefault(sd => sd.ServiceType == typeof(DbContextOptions<TContext>));
+
+            if (oldDbContextOptionsDescriptor is null)
+            {
+                throw new InvalidOperationException($"DbContext<{typeof(TContext).Name}> was not registered");
+            }
+
+            builder.Services.Remove(oldDbContextOptionsDescriptor);
+
+            var dbContextOptionsDescriptor = new ServiceDescriptor(
+                oldDbContextOptionsDescriptor.ServiceType,
+                oldDbContextOptionsDescriptor.ServiceKey,
+                factory: (sp, key) =>
+                {
+                    var dbContextOptions = oldDbContextOptionsDescriptor.ImplementationFactory?.Invoke(sp) as DbContextOptions<TContext>;
+
+#pragma warning disable EF1001 // Internal EF Core API usage.
+                    if (dbContextOptions is null ||
+                        dbContextOptions.FindExtension<MySqlOptionsExtension>() is not MySqlOptionsExtension extension ||
+                        extension.ServerVersion is not ServerVersion serverVersion)
+                    {
+                        throw new InvalidOperationException($"A DbContextOptions<{typeof(TContext).Name}> was not found. Please ensure 'ServerVersion' was configured.");
```

Comment:
I wonder if users will know what "Please ensure 'ServerVersion' was configured." means. Maybe it would be better to say

```suggestion
                        throw new InvalidOperationException($"A DbContextOptions<{typeof(TContext).Name}> was not found. Please ensure 'UseMySql' was called correctly with a ServerVersion.");
```